app/services/cache_service.rb

```diff
@@ -43,7 +47,34 @@ def listing_unchanged?(prev_cached_listing, fresh_listing)
       (prev_cached_listing['LastModifiedDate'] == fresh_listing['LastModifiedDate'])
   end
 
+  def listing_images_unchanged?(prev_cached_listing, fresh_listing)
+    prev_cached_listing_images = prev_cached_listing.present? && prev_cached_listing['Listing_Images']
+    fresh_listing_images = fresh_listing.present? && fresh_listing['Listing_Images']
+
+    return true if fresh_listing_images.blank?
+
+    Rails.logger.info("#{fresh_listing['Id']}: Listing_Images length: Prev: #{prev_cached_listing_images&.length} Fresh: #{fresh_listing_images&.length}")
+
+    return false if prev_cached_listing_images&.length != fresh_listing_images&.length
+
+    unchanged = true
+    fresh_listing_images.each_with_index do |li, idx|
+      unchanged = listing_image_unchanged?(prev_cached_listing_images[idx], li)
+      break unless unchanged
+    end
+    Rails.logger.info("Listing Images for  #{fresh_listing['Id']} is unchanged: #{unchanged}")
+    unchanged
+  end
```

Comment:
There might be a slight performance hit but this method could be more flexible and easier to read if we compare arrays of key-value pairs.

```ruby
def foobar(fresh_listing_images, prev_cached_listing_images)
  foo = fresh_listing_images.map { |li| li.slice('Id', 'Image_URL') }
  bar = prev_cached_listing_images.map { |li| li.slice('Id', 'Image_URL') }
  Rails.logger.info("foo #{foo}, bar #{bar}")
  (foo - bar).empty? && (bar - foo).empty?
end
```